### ISSUE TITLE
Investigating ...

### ISSUE THREAD

**Issue by fwyzard** [vkhristenko/repr_cuda_build_problems#1]
I'm creating an issue to write down things as I look into this.

The first observation is that linking with `lib1.$(LIB_EXTENSION)` here:
https://github.com/vkhristenko/repr_cuda_build_problems/blob/2c61c780d4c44e8bb19f54f17c882cacb3e2071b/Makefile#L21
makes no difference; if I remove it I still get the same error.

**Comment by vkhristenko** [comment]
Yep, i added it for completeness, as this is what you would see in `cmssw`. 

the real reason to include this is for the case when you do not load lib1 first, you will get `undefined symbol` upon loading lib2. in other words, if you link on the device, you still need host side sections... you can not just link device (-dlink) and not to include host side sections...

**Comment by fwyzard** [comment]
Right, I see:
```
Cannot open lib: /home/fwyzard/test/cuda_build_problems/lib1.so problem: /home/fwyzard/test/cuda_build_problems/lib2.so: undefined symbol: __fatbinwrap_42_tmpxft_000280ad_00000000_6_kernels_cpp1_ii_ad0b978c
```
